### PR TITLE
gilbert: surface-decoder-only FiLM normal conditioning for wsy/wsz (Wave 14 R2)

### PR DESCRIPTION
**Pull request by morganmcg1**
# Hypothesis: Surface-decoder-only FiLM normal conditioning for wsy/wsz (Wave 14 reassignment R2)

**Reassignment context**: This is a reassignment of closed PR #441, which expired its 30-min ACK deadline. The hypothesis is unchanged. **Strict 30-min ACK deadline applies — post your W&B run ID within 30 minutes of receiving this PR or it will be closed and the work permanently reassigned to another student.**

## Why this hypothesis

The current best baseline (alphonse PR #174, val_abupt = 6.9549%) is bottlenecked by wall-shear axes:
- wsy = 8.7345% (target 3.65% — **5.08pp gap**)
- wsz = 10.5766% (target 3.63% — **6.95pp gap**)

The previous FiLM-every-block experiment (gilbert PR #346) **falsified trunk-level normal conditioning**: ep10 abupt looked OK (8.86%) but wsy/wsz were ~2pp WORSE than baseline. Diagnosis: FiLM applied across every block in the shared trunk diluted volume-pressure capacity into surface-only conditioning, hurting overall axis balance.

**This experiment isolates the conditioning to the surface decoder only**, leaving the shared trunk untouched. Hypothesis: surface normals are physically meaningful for wall-shear prediction (shear is tangential to the surface), but the conditioning signal needs to be applied where surface features are read out — not in the shared backbone.

## Implementation

Add a `--film-surface-decoder` flag. When enabled:

1. **Conditioning input**: Surface normal vector at each point (3-dim). Surface normals are already computed and available in the surface batch.
2. **FiLM generator MLP**: `Linear(3 → 64) → GELU → Linear(64 → 2 * hidden_dim)`, output split into `gamma` and `beta`. **Initialize the output Linear's weights and biases to zero** so γ=0, β=0 at init → identity FiLM (gamma=1+0, beta=0). This gives a clean ablation: at step 0 the network is identical to the baseline.
3. **Where to apply**: Inside the surface decoder MLP only. The DrivAerML model has a separate surface decoder head — apply FiLM after each linear+activation in that head: `h = h * (1 + gamma) + beta`. **Do NOT touch the shared trunk or the volume decoder.** Volume pressure prediction must be unchanged.
4. **No FiLM in volume decoder**: This is the critical contrast vs. PR #346.

## Reproduce command

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --fourier-pe \
  --no-use-ema \
  --lr 3e-4 \
  --lr-cosine-t-max 50 \
  --no-compile-model \
  --film-surface-decoder \
  --agent gilbert \
  --wandb-group bengio-wave14 \
  --wandb-name gilbert-surface-dec-film-normal-r2-5l256d-tmax50
```

## Gate schedule (kill if NOT below)

| Epoch | val_abupt kill threshold |
|-------|--------------------------|
| ep5   | < 13.0%                  |
| ep10  | < 10.0%                  |
| ep15  | < 8.9%                   |
| ep30  | < 7.5%                   |
| ep50  | must beat 6.9549%        |

**Critical reporting requirement**: At each gate, report `val_primary/wall_shear_y_rel_l2_pct` and `val_primary/wall_shear_z_rel_l2_pct` separately. The hypothesis succeeds only if wsy/wsz improve materially vs the alphonse PR #174 baseline (wsy 8.7345%, wsz 10.5766%) — even if abupt is similar, a clear binding-axis improvement at zero cost to vol_p (must remain ≤ 4.5%) is mergeable.

## Hard constraints

- `--no-compile-model`: Mandatory (PyTorch 2.x Inductor crash at validation)
- `--fourier-pe`: Mandatory (n_params confirms FourierEmbed presence — comparability)
- Correct grad-clip flag if you tweak it: `--grad-clip-norm` (NOT `--grad-clip`)
- Volume decoder must remain identical to baseline — verify via grad flow if uncertain
- Epochs hard cap: `SENPAI_MAX_EPOCHS`; wall-clock: `SENPAI_TIMEOUT_MINUTES`

## Baseline metrics (alphonse PR #174, run vu4jsiic)

| Metric | val |
|--------|-----|
| val_primary/abupt_axis_mean_rel_l2_pct | **6.9549** |
| val_primary/surface_pressure_rel_l2_pct | 4.5644 |
| val_primary/volume_pressure_rel_l2_pct | 3.9361 ✓ |
| val_primary/wall_shear_y_rel_l2_pct | 8.7345 |
| val_primary/wall_shear_z_rel_l2_pct | 10.5766 |

## ACK requirement (HARD)

Post a comment on this PR within **30 minutes** of receipt with:
1. Your W&B run ID
2. Confirmation that training has started (the first epoch is in progress)
3. Confirmation that the `--film-surface-decoder` flag was correctly applied (only in surface decoder, NOT trunk or volume decoder)

Failure to ACK will result in closure and permanent reassignment.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: **ACK deadline approaching — 12 minutes remaining (by 19:19Z).** No response received yet on PR #447 (surface-decoder-only FiLM normal conditioning). Start your run immediately and post your W&B run ID. Note: you missed your Wave 14 first ACK — a second miss will result in this assignment being reallocated.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #447 because ADVISOR: PR #447 CLOSED — ACK deadline (19:19Z) missed. This is gilbert's **second consecutive ACK failure** (also missed Wave 14 first ACK). Per lab policy: two consecutive misses = automatic reallocation. This assignment (surface-decoder-only FiLM normal conditioning for wsy/wsz) is being reassigned to another student. gilbert, if you return to active status, please reach out to the advisor for a new assignment..
